src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -0,0 +1,749 @@
+                _length = result._length;
+            }
+
+            public void Multiply10()
+            {
+                if (IsZero())
+                {
+                    return;
+                }
+
+                int index = 0;
+                int length = _length;
+                ulong carry = 0;
+
+                while (index < length)
+                {
+                    var block = (ulong)(_blocks[index]);
+                    ulong product = (block << 3) + (block << 1) + carry;
+                    carry = product >> 32;
+                    _blocks[index] = (uint)(product);
+
+                    index++;
+                }
+
+                if (carry != 0)
+                {
+                    Debug.Assert(unchecked((uint)(_length)) + 1 <= MaxBlockCount);
+                    _blocks[index] = (uint)(carry);
+                    _length += 1;
+                }
+            }
+
+            public void SetUInt32(uint value)
+            {
+                _blocks[0] = value;
+                _length = 1;
+            }
+
+            public void SetUInt64(ulong value)
+            {
+                var lower = (uint)(value);
+                var upper = (uint)(value >> 32);
+
+                _blocks[0] = lower;
+                _blocks[1] = upper;
+
+                _length = (upper == 0) ? 1 : 2;
+            }
+
+            public void SetValue(ref BigInteger rhs)
+            {
+                int rhsLength = rhs._length;
+                Buffer.Memcpy((byte*)(_blocks.GetPointer()), (byte*)(rhs._blocks.GetPointer()), (rhsLength * sizeof(uint)));
+                _length = rhsLength;
+            }
+
+            public void SetZero()
+            {
+                _length = 0;
+            }
+
+            public void ShiftLeft(uint shift)
+            {
+                // Process blocks high to low so that we can safely process in place
+                var length = _length;
+
+                if ((length == 0) || (shift == 0))
+                {
+                    return;
+                }
+
+                uint blocksToShift = Math.DivRem(shift, 32, out uint remainingBitsToShift);
+
+                // Copy blocks from high to low
+                int readIndex = (length - 1);
+                int writeIndex = readIndex + (int)(blocksToShift);
+
+                // Check if the shift is block aligned
+                if (remainingBitsToShift == 0)
+                {
+                    Debug.Assert(writeIndex < MaxBlockCount);
+
+                    while (readIndex >= 0)
+                    {
+                        _blocks[writeIndex] = _blocks[readIndex];
+                        readIndex--;
+                        writeIndex--;
+                    }
+
+                    _length += (int)(blocksToShift);
+
+                    // Zero the remaining low blocks
+                    Buffer.ZeroMemory((byte*)(_blocks.GetPointer()), (blocksToShift * sizeof(uint)));
```

Comment:
This was not being done properly in native code (size was `blocksToShift`, rather than `blocksToShift * sizeof(uint)`).

Without `Grisu3` enabled as the first code-path, the Roslyn `RealParser` tests exposed this issue.